lib/new_relic/error/logger_handler.ex

```diff
@@ -1,30 +1,33 @@
 defmodule NewRelic.Error.LoggerHandler do
   @moduledoc false
 
-  # http://erlang.org/doc/man/logger.html
-
   def add_handler() do
-    :logger.remove_handler(:new_relic)
-    :logger.add_handler(:new_relic, NewRelic.Error.LoggerHandler, %{})
+    Logger.add_translator({__MODULE__, :translator})
   end
 
   def remove_handler() do
-    :logger.remove_handler(:new_relic)
+    Logger.remove_translator({__MODULE__, :translator})
   end
 
-  def log(
-        %{
-          meta: %{error_logger: %{tag: :error_report, type: :crash_report}},
-          msg: {:report, %{report: [report | _]}}
-        },
-        _config
-      ) do
+  def translator(_level, _message, _timestamp, {_, [report | _]}) when is_list(report) do
     if NewRelic.Transaction.Sidecar.tracking?() do
       NewRelic.Error.Reporter.report_error(:transaction, report)
     else
       NewRelic.Error.Reporter.report_error(:process, report)
     end
+
+    :none
+  end
+
+  def translator(_level, :error, _timestamp, {_, %{args: _, function: _}} = metadata) do
```

Comment:
@tmaszk I looked more closely at 1.14, and 1.15.7 for this: https://hexdocs.pm/logger/1.15.7/Logger.Translator.html#callbacks

We may have gotten it slightly wrong. I think the args are `translator(_min_level, _level, :error, report_tuple)`. And it seems like `kind` is either `:error` or `:report` explicitly.

Reply:
Correction :format or :report, so :error is the level being reported. Ignore me!